src/libAtomVM/externalterm.c

```diff
@@ -209,6 +232,19 @@ static int serialize_term(Context *ctx, uint8_t *buf, term t)
         }
         return 2;
 
+    } else if (term_is_boxed_integer(t) || (term_is_integer(t) && (term_to_int(t) > INT32_MAX || term_to_int(t) < INT32_MIN))) {
+        // TODO For now, the max integer size we support is 64 bits
+        avm_int64_t val = term_maybe_unbox_int64(t);
+        avm_uint64_t unsigned_val = (val < 0) ? val * -1 : val;
```

Comment:
I am not sure it is reasonable to make those checks, since we don't (currently) support boxed integers > 64 bits.

Reply:
> any reason why we are doing `* -1` instead of `-val`?

Changed it to -val.  I am not sure the compiler would not do anyway, but I made the fix.

Reply:
correct me if I am wrong, but a boxed integer can store MAX_INT64 and MIN_INT64 which are the lowest and the highest (still) supported values.
However they are still tricky to handle so we should check them. i.e. MAX_INT64 = 9223372036854775807 and MIN_INT64 = -9223372036854775808 so -MIN_INT64 > MAX_INT64.

Reply:
I see.  Since this is a conversion from a boxed int, and then a write into an unsigned, I think the check actually should go in the deserialized.  See the change to the parsing code below (actually, in the size calculation, since that is where we raise errors in parsing)